### ISSUE TITLE
Connectivity test fails on GKE cluster with encryption enabled

### ISSUE THREAD

**Issue by nebril** [cilium/cilium#15434]
## Bug report

**General Information**

- Cilium version (run `cilium version`): `1.9.4`
- Kernel version (run `uname -a`): `Linux gke-test-cluster-default-pool-ca6cdc60-49jx 5.4.89+ #1 SMP Thu Jan 28 22:59:01 PST 2021 x86_64 x86_64 x86_64 GNU/Linux`
- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...):
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.15-gke.1501", GitCommit:"de67bb0d58413ba2ba9b64810ab438a9734a2ab9", GitTreeState:"clean", BuildDate:"2021-02-26T23:41:52Z", GoVersion:"go1.13.15b4", Compiler:"gc", Platform:"linux/amd64"}
```

Sysdump: [cilium-sysdump-out.zip.zip](https://github.com/cilium/cilium/files/6189924/cilium-sysdump-out.zip.zip)

**How to reproduce the issue**

1. create GKE cluster and setup kubectl
```
          gcloud container clusters create clustername \
            --preemptible \
            --image-type COS \
            --num-nodes 2 \
            --machine-type n1-standard-4 \
            --zone us-west2-a

          gcloud container clusters get-credentials clustername --zone us-west2-a
```
2. get `cilium-cli`
```
          curl -LO https://github.com/cilium/cilium-cli/releases/download/v0.4/cilium-linux-amd64.tar.gz
          sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/bin
          rm cilium-linux-amd64.tar.gz
```
3. install Cilium in the cluster with encryption enabled
```
          cilium install --encryption
          cilium status --wait
```
4. run connectivity test
```
          cilium connectivity test
```

**Comment by nebril** [comment]
The exact same thing is happening on master

**Comment by nbusseneau** [comment]
Maybe it will be fixed by #15357?

**Comment by jrfastab** [comment]
Resolved by #15357.
